packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -0,0 +1,272 @@
+          dependencyType: "peerDependencies",
+          dependencyName: peerDependencyName,
+          version: mostStrictPeerRequirement.range,
+        }),
+      });
+    }
+
+    if (rootPeerDependencyRange != null && !doesASatisfyB(rootPeerDependencyRange, mostStrictPeerRequirement.range)) {
+      context.addError({
+        file: rootPackageJsonPath,
+        message: `[4] Package ${rootPackageName} peer dependency on ${peerDependencyName} is not strict enough.`,
+        longMessage: `Dependency ${mostStrictPeerRequirement.node.packageJson.name} requires ${mostStrictPeerRequirement.range} or stricter.`,
+        fixer: getAddDependencyTypeFixer({
+          packageJsonPath: rootPackageJsonPath,
+          dependencyType: "peerDependencies",
+          dependencyName: peerDependencyName,
+          version: mostStrictPeerRequirement.range,
+        }),
+      });
+    }
+  }
+}
+
+/*
+  Returns `true` if `a` is more strict than or equal to `b`
+  Returns `false` otherwise
+
+  This code assumes that major versions are not repeated in union ranges.
+  e.g. `^15.0.5 || ^16.0.0`, but not `15.0.5 || 15.0.999`
+
+  To determine that `a` is "more strict than or equal to" `b`:
+  For set `a` of version values (`15.0.5`) and ranges (`^16.0.0`),
+  each entry must be permitted by (equal to or greater than) some entry in `b`.
+
+  Actual dependencies of `a` can be more restrictive along matching `^`? majors in `b`, but not less.
+  For example, the following values are more strict than or equal to
+  (satisfy a required dependency) `^15.0.5 || ^16.0.0`:
+  - `^15.0.5 || ^16.0.0`
+  - `^15.0.5 || ^16.x.x`
+  - `^15.0.5 || ^16`
+  - `15.0.5 || 16.0.0`
+  - `^15.0.999 || ^16.0.0`
+  - `^15.0.5 || ^16.0.999`
+  - `^15.0.999`
+  - `^16.0.0`
+  - `16.0.0`
+  Here are some sample values that are less strict than (do not satisfy a required dependency) `^15.0.0 || ^16.0.0`:
+  - `^15.0.5 || ^16.0.0 || ^17.0.0`
+  - `^14.0.0 || ^15.0.5 || ^16.0.0`
+  - `^15.0.0 || ^16.0.0`
+  - `^17.0.0`
+  - `17.0.0`
+*/
+export function doesASatisfyB(a: string, b: string): boolean {
+  const aIsAnyVersionRange = isAnyVersionRange(a);
+  const bIsAnyVersionRange = isAnyVersionRange(b);
+  if (bIsAnyVersionRange) {
+    return true;
+  } else if (aIsAnyVersionRange) {
+    // `bIsAnyVersionRange` is `false`
+    // `a` permits more values than `b`, therefore `a` is "less strict"
+    return false;
+  }
+
+  const aVersions = a.includes("||") ? a.split("||").map(s => s.trim()) : [a];
+  const bVersions = b.includes("||") ? b.split("||").map(s => s.trim()) : [b];
+
+  return aVersions.every(aVersion => {
+    const aMajor = coerce(aVersion)!.major;
+    const majorMatchingBVersion = bVersions.find(m => coerce(m)!.major === aMajor);
+    if (majorMatchingBVersion === undefined) {
+      // `a` permits a major version that is not permitted by `b`, therefore `a` is "less strict"
+      return false;
+    }
+
+    const aSemVer = coerce(aVersion)!;
```

Comment:
why do we coerce twice?

Reply:
i preferred readability over whatever minor perf different there might be
happy to change it tho